### PR TITLE
Extensions-api, new nav-functions, route-api 

### LINKED ISSUE #72: Chained navigation request
Be able to call to set specific path from navController including sub-navControllers

EG: 
navController.setPath(Root).then(Tabs).then(HomeTab,ProductTab).then(ProductDetails)
-->
1st nc opens Root screen
Root's screen inner NC(navController) opens Tabs 
Tabs's NC put HomeTab in backstack and opens ProductTab
ProductTab opens ProductDetails

Pros: 
 - easy to use with complex navigation
 
Cons: 
 - will need to make dsl to generate tree
 - can not controll subtrees? (eg: what if i want to put smth inside HomeTab's NC)

### LINKED ISSUE #74: Add option to operate with NavEntry in the 'navigate' 'replace' edit-backstack
EG:
```kotlin
val t = SomeScreen.toNavEntry(freeArgs=1)
navController.navigate(t)
```

### LINKED ISSUE #88: Improve route api
Add smth similar to

NavController.route(vararg RouteElement)
NavController.route(screenA,screeB,navControllerC, screenD, screemE)

options: 
    fail strat
    replace-logic

RouteElement:
     a: destination-resolver (find from list of destinations)
     b: destination
     c: nc-resolver/key  (empty => take first)

destination: add option to pass nav/free args

realization -> build a route similar to current api base on trace

### PR DESCRIPTION
**Pull request by vkatz**
- closes #74
- closes #72
- closes #88

### CODE REVIEW

**Review by egorikftp — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -272,7 +299,87 @@ Hint
...
+            )
+            clearFreeArgs()
+        }
+    }
+
+    Navigation(modifier = Modifier.fillMaxSize(), navController = deeplinkNavController)
+}
+
+//---- idea 2 -----
+// use rout-api
```
[the remark this answers is not in the document]
Comment (vkatz) [reply README.md:339]: fixed

File: tiamat/src/commonMain/kotlin/com/composegears/tiamat/Utils.kt [hunk 2]
```diff
@@ -14,4 +14,8 @@ internal fun <T> Iterable<T>.allIndexed(predicate: (Int, T) -> Boolean): Boolean
         if (!predicate(index, element)) return false
     }
     return true
-}
\ No newline at end of file
+}
+
+@Retention(AnnotationRetention.BINARY)
+@RequiresOptIn(level = RequiresOptIn.Level.WARNING, message = "Under development feature")
```
[the remark this answers is not in the document]
Comment (vkatz) [reply tiamat/src/commonMain/kotlin/com/composegears/tiamat/Utils.kt:20]: I'v used `This is an experimental Tiamat API, and it is likely to be changed in the future.` as we dont have plans to remove it 100%

File: tiamat/build.gradle.kts [hunk 3]
```diff
@@ -10,7 +10,7 @@ plugins {
     alias(libs.plugins.m2p)
 }
 
-version = "1.2.0"
+version = "1.2.1"
```
[the remark this answers is not in the document]
Comment (vkatz) [reply tiamat/build.gradle.kts:13]: agree

**Comment by vkatz** [comment]
Too complex builder, will rework it base on #88 idea

**Pull request merged**